src/main/java/com/minecolonies/colony/Colony.java

```diff
@@ -817,7 +810,7 @@ public void removeBuilding(Building building)
             townHall = null;
         }
 
-        //  Allow Citizens to fix up any data that wasn't fixed up by the Building's own onDestroyed
+        //  Allow Citizens to fix up any data that wasn't fixed up by the AbstractBuilding's own onDestroyed
         for (CitizenData citizen : citizens.values())
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Local variable 'citizen' could be declared final [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3ALocalVariableCouldBeFinal)